src/d3/hapticplot.d3.ts

```diff
@@ -0,0 +1,137 @@
+// run `tsc Hapticplot.ts` to compile into Hapticplot.js file
+
+// for running on browser
+// import * as _d3 from 'd3';
+
+// declare global {
+// const d3: typeof _d3;
+// }
+
+// for running on unit tests
+import * as d3 from 'd3';
+import * as THREE from 'three';
+
+export class Hapticplot{
+    private data: number[];
+    private shape: string;
+    private container: HTMLElement | null;
+    private hscale: d3.ScaleLinear<number, number>;
+
+  constructor(shape: string) {
+    this.shape = shape;
+  }
+  init(container: HTMLElement | null, data: number[]){
+    this.data = data;
+    this.container = container;
+    // create a scale so that there is correspondence between data set and screen render
+    this.hscale = d3.scaleLinear();
+    this.hscale.domain([0, d3.max(this.data) as number])       // max of dataset
+    .range([0, 100]);                                      // linear mapping of data set values to values from 0 to 10
+    this.createSky();
+    this.generatePts();
+    this.setColor('blue');
+    this.setRadius(0.1);
+    this.setInteraction();
+    this.addEventListeners();
+    this.createGridPlane();
+  }
+
+  private generatePts() {
+     /*
+     Generates points based on initilization data
+      - "enter" identifies any DOM elements to be added when # array elements & # DOM elements don't match
+      - select all generated entities and updates their position based on ingested data
+    */
+    d3.select(this.container).selectAll(this.shape).data(this.data).enter().append(this.shape);
```

Comment:
Typically prefer selecting on css class, rather than element name.  It's been shown to be much faster.

Reply:
I switched the selectAll method to use the css class "datapoint," but I'm having issues when I try to switch the container to a css selection

Reply:
I should have been more specific; I just meant the shape.  The container can stay as it is, since it's already got the element itself